src/shared/backend/services/cloudbuild.ts

```diff
@@ -88,6 +88,37 @@ export const createJob = async (params: JobStatusParams): Promise<Job> => {
   return data as Job;
 };
 
+// Query a job, if it's not found, create it
+export const queryCreateJob = async (params: JobStatusParams): Promise<Job> => {
```

Comment:
IMO I don't like this function. Does the API let you create a job if it already exists? It might be best to run `createJob` and if it already exists ignore the error. This should all be done in the main job code so it's clear what's going on. This function is too confusing basically